### ISSUE TITLE
Questions and Feedback on the GSuite Exporter 

### ISSUE THREAD

**Issue by yhuang** [GoogleCloudPlatform/professional-services#217]
**Questions**

1.  Are more G Suite Admin APIs supported now?  The documentation indicates activity reports for admin, drive, login, mobile, and token are supported, but I see in the error message that the script expects other activities:

```
<HttpError 400 when requesting https://www.googleapis.com/admin/reports/v1/activity/users/all/applications/login%20token?alt=json returned "Invalid value 'login token'. Values must match the following regular expression: '(admin)|(calendar)|(drive)|(login)|(mobile)|(token)|(groups)|(saml)|(chat)|(gplus)|(rules)|(jamboard)|(meet)|(user_accounts)|(access_transparency)'
```

2.  Are there example screenshots of what the exported logs would look like?

3.  If the script runner is expected to authenticate with his or her G Suite / Cloud Identity Super Admin user account, how would we run this export command periodically as a cron job via a GCE instance?  Can the script runner be a service account?  If so, how would the authentication process look?

4.  When I tried to use a service account as the argument for `--admin-user`, I got the following error:
```
unauthorized_client: Client is unauthorized to retrieve access tokens using this method, or client not authorized for any of the scopes requested.'
```
How would the proper authorization be granted to a service account we want to use?

**Feedback**

1.  One of the listed requirements is *NOT* accurate.  `roles/iam.tokenCreator` should really be `roles/iam.serviceAccountTokenCreator`.

2.  This example as is would yield an error for the user, as the `--applications` option cannot handle space delimited words.  To export additional activities, I had to run the gsuite-exporter command on individual APIs separately.

```
gsuite-exporter
  --credentials-path='/path/to/service/account/credentials.json'
  --admin-user='<your_gsuite_admin>@<your_domain>'
  --api='report_v1'
  --applications='login drive token'
  --project-id='<logging_project>'
  --exporter='stackdriver_exporter.StackdriverExporter'
```

3. The process of configuring G Suite admin account should be better documented.  I had to scour the web to find documentation on a [third-party site](https://www.fin.com/posts/navigating-the-google-suite-directory-api/).  Give how different G Suite is from GCP, please invest the time to explain in plain English why each step in the setup is required.

4.  Is there a reason this Python package is only limited to Python 2.7?  Please consider getting it working for Python 3 as well.

5.  Please explain the purpose of each parameter.  The repo currently does not provide any documentation on the required parameters (`--credentials-path`, `--admin-user`, `--api`, `--applications`, `--project-id`, and `--exporter`).  The lack of documentation on the parameters is especially frustrating, as a naive user would not know immediately why both `--credentials-path` and `--admin-user` are required and that the `--admin-user` is expected to be the G Suite / Cloud Identity Super Admin human user.  Given that expectation, the README should also explain how the authentication of the Super Admin should be done.  To get the script to run, I figured out that I needed to run `gcloud auth login` first to authenticate with my Super Admin identity.  That detail should be provided in the README clearly.

**Comment by ocervell** [comment]
@yhuang,

Thanks for your feedback. You are making good points that the documentation can be updated to explain what each parameter does in detail. Please remember that although we make such tools to help users, partners and customers, they are still in experimental stage (definitely not production) and we provide limited support for those. That said, we take your feedback seriously and thrive to improve the existing code and documentation.

As for your questions:

1. At the moment, only the APIs listed in the README are supported. The message you're seeing comes from the Google APIs, not the script itself.

2. The exported logs will be available in the project you're syncing too, under 'global' log type in Stackdriver Logging. They look like normal JSON-formatted logs, [here](https://user-images.githubusercontent.com/9629314/54856793-3d79ad80-4cca-11e9-9e8d-3c5fdb80e587.png) is an example of a log record.

3. To run the script as a cronjob, I recommend using the [terraform-google-gsuite-export](https://github.com/terraform-google-modules/terraform-google-gsuite-export) module that configures it automatically for you.

4. You cannot use a service account for `admin-user`. A GSuite Admin account is required, as specified in the README under 'Requirements' section. A service account cannot be a GSuite Administrator for security reasons. Service accounts can only 'impersonate' GSuite Administrators using GSuite token-based security (client scopes), which is why we require both the `service-account` and `admin-user` fields.

**Comment by agold-rh** [comment]
Closed due to inactivity.
